marketplace/google-analytics-app/src/Timeline.tsx

```diff
@@ -0,0 +1,132 @@
+/* global gapi */
+
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+import { formatDate } from './utils';
+import styles from './styles';
+import {
+  SkeletonImage,
+  TextLink,
+  SkeletonContainer,
+  Paragraph
+} from '@contentful/forma-36-react-components';
+
+const CHART_HEIGHT = 200;
+
+export default class Timeline extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      timeline: null,
+      viewUrl: null
+    };
+
+    this.onSuccess = ({ data }) => this.props.onData(data);
+  }
+
+  shouldComponentUpdate(nextProps) {
+    const { start, end } = this.props.range;
+    const nextRange = nextProps.range;
+
+    return !this.state.timeline || nextRange.start !== start || nextRange.end !== end;
+  }
+
+  async componentDidMount() {
+    let viewUrl = '';
+
+    try {
+      const accounts = (await gapi.client.analytics.management.accountSummaries.list()) || [];
+      viewUrl = this.getExternalUrl(accounts);
+      // console.log(accounts)
+    } catch (error) {
+      console.log(error);
```

Comment:
We should probably have a better UI for this with either showing an error message in the sidebar or using our notifier.

```suggestion
  this.props.sdk.notifier.error('Failed to load your Google Analytics data');
```